base_mixin_restrict_field_access/models/restrict_field_access_mixin.py

```diff
@@ -72,6 +72,61 @@ def read(self, fields=None, load='_classic_read'):
                         self._fields[field].null(self.env))
         return result
 
+    def read_group(self, cr, uid, domain, fields, groupby, offset=0, limit=None, context=None, orderby=False, lazy=True):
+        """
+        Remove inaccessible fields from 'fields', 'groupby' and 'orderby'.
+
+        If this removes all 'fields', return no records.
+        If this removes all 'groupby', group by first remaining field.
+        If this removes 'orderby', don't specify order.
+        """
+        sanitised_fields = [f for f in fields if self._restrict_field_access_is_field_accessible(cr, uid, [], f)]
```

Comment:
You're right I'll adjust so fields default to all columns (before getting sanitised).

Determining dynamically is still possible, you just need to decide what it means when self is falsy. For our HR module we use this:

`
    @api.multi
    def _restrict_field_access_is_field_accessible(self, field_name,
                                                   action='read'):
        """Determine field access dynamically."""
        if field_name in FIELDS_ONLY_MANAGER:
            # "HR Settings" tab: HR Manager only
            return self.is_hr_manager()
        elif field_name in FIELDS_ONLY_SELF_OR_MANAGER:
            # "Personal Information" tab: HR Manager or Employee's user only
            if self.is_hr_manager():
                return True
            else:
                if not self:
                    # Not called on particular instance (e.g. export)
                    return self.is_hr_manager()
                else:
                    return all(this.user_id == self.env.user for this in self)
        else:
            # "Public Information" tab
            if action == 'read':
                # Read only: no field restriction
                return True
            else:   # 'create'/'write'
                # Create/Write: HR Manager or Employee's user only
                if self.is_hr_manager():
                    return True
                else:
                    if not self:
                        # Not called on particular instance
                        return self.is_hr_manager()
                    else:
                        return all(this.user_id == self.env.user for this in self)
`

I've got something for using null values, but I'm not sure if I like it. They all go into an "Undefined" group, also not sure if count zero makes sense. Will update the PR soon.